.github/workflows/ci-matrix-build.yml

```diff
@@ -0,0 +1,69 @@
+#This workflow runs matrix build with both versions of setup MATLAB action hourly
+
+name: MATLAB matrix build
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [main, setup-matlab-matrix-build]
+  pull_request:
+    branches: [main]
+  # Schedule the workflow to run hourly
+  schedule:
+    - cron: "0 * * * *"
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  # This workflow contains a single job called "build-v1"
+  build-v1:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      # Sets up MATLAB on the GitHub Actions runner
+      - name: Setup MATLAB
+        uses: matlab-actions/setup-matlab@v1
+
+      # Runs a set of commands using the runners shell
+      - name: Run tests and generate artifacts
+        uses: matlab-actions/run-tests@v1
+        with:
+          source-folder: code
+          test-results-junit: test-results/results.xml
+          code-coverage-cobertura: code-coverage/coverage.xml
+
+  build-v2-beta:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      # Sets up MATLAB on the GitHub Actions runner
+      - name: Setup MATLAB
+        uses: matlab-actions/setup-matlab@v2-beta
+
+      # Runs a set of commands using the runners shell
+      - name: Run tests and generate artifacts
+        uses: matlab-actions/run-tests@v1
+        with:
+          source-folder: code
+          test-results-junit: test-results/results.xml
+          code-coverage-cobertura: code-coverage/coverage.xml
+
+      # As an alternative to run-tests, you can use run-command to execute a MATLAB script, function, or statement.
+      #- name: Run all tests
+      #  uses: matlab-actions/run-command@v1
+      #  with:
+      #    command: addpath('code'); results = runtests('IncludeSubfolders', true); assertSuccess(results);
```

Comment:
I think this is more of a doc note for end users. We also lost the notification on fails at the bottom of the hourly branch. I think we might want to keep that, especially if we end up merging this into hourly.